src/SignalR/common/Http.Connections/src/Internal/HttpConnectionManager.cs

```diff
@@ -138,17 +133,19 @@ private async Task ExecuteTimerLoop()
 
         public void Scan()
         {
+            var now = DateTimeOffset.UtcNow;
+
             // Scan the registered connections looking for ones that have timed out
             foreach (var c in _connections)
             {
                 var connection = c.Value.Connection;
                 // Capture the connection state
-                var lastSeenUtc = connection.LastSeenUtcIfInactive;
+                var lastSeenTick = connection.LastSeenTicksIfInactive;
 
-                var utcNow = DateTimeOffset.UtcNow;
+                var ticks = Environment.TickCount64;
```

Comment:
Is this loop really so slow that we cant evaluate this outside the loop like you did with `now`?

Reply:
```suggestion
```